src/packet/signature.js

```diff
@@ -133,15 +131,27 @@ class SignaturePacket {
     this.signedHashValue = bytes.subarray(i, i + 2);
     i += 2;
 
-    this.signature = bytes.subarray(i, bytes.length);
+    this.signature = crypto.signature.parseSignatureParams(this.publicKeyAlgorithm, bytes.subarray(i, bytes.length));
+  }
+
+  /**
+   * @returns {Uint8Array | ReadableStream<Uint8Array>}
+   */
+  write_signature() {
+    if (util.isPromise(this.signature)) {
```

Comment:
Perhaps `this.params instanceof Promise`? Then you don't need to add a helper function just for this.

Reply:
This is a corner case but with some libraries/polyfill promises aren't `Promise`s. Checking if something is a `thenable` is less prone to errors even when non-standard Promises exist.

Reply:
Yeah, but.. this Promise is coming from calling an `async` function without `await`, so it can only be a built-in Promise. If someone transpiles the code for an environment where Promises don't exist, then `Promise` should also point to the polyfilled `Promise` implementation.

Reply:
Removed the util.

There's a non-0 percent chance of this happening though: https://xkcd.com/1172/ :D